tasks/run-file-updates/tests/mocks.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+set -eux
+
+# mocks to be injected into task step scripts
+
+function internal-request() {
+  TIMEOUT=30
+  END_TIME=$(date -ud "$TIMEOUT seconds" +%s)
+
+  echo Mock internal-request called with: $*
+  echo $* >> $(workspaces.data.path)/mock_internal-request.txt
+
+  # since we put the IR in the background, we need to be able to locate it so we can
+  # get the name to patch it. We do this by tacking on another random label that we can use
+  # to select with later.
+  rando=$(openssl rand -hex 12)
+  /home/utils/internal-request "$@" -l "internal-services.appstudio.openshift.io/test-id=$rando" &
+
+  sleep 2
+  NAME=
+  while [[ -z ${NAME} ]]; do
+    if [ "$(date +%s)" -gt "$END_TIME" ]; then
+        echo "ERROR: Timeout while waiting to locate InternalRequest"
+        echo "Internal requests:"
+        kubectl get internalrequest --no-headers -o custom-columns=":metadata.name" \
+            --sort-by=.metadata.creationTimestamp
+        exit 124
+    fi
+
+    NAME=$(kubectl get internalrequest -l "internal-services.appstudio.openshift.io/test-id=$rando" \
+        --no-headers -o custom-columns=":metadata.name" \
+        --sort-by=.metadata.creationTimestamp | tail -1)
+    if [ -z $NAME ]; then
+        echo "Warning: Unable to get IR name"
+        sleep 2
+    fi
+  done
+  echo "IR Name: $NAME"
+
+  if [[ "$*" == *"requester=testuser-failure"* ]]; then
```

Comment:
This is copied from rh-sign-image. The requester param isn't used in this task, so you can remove lines 40-46 except line 45 I think

Reply:
fixed